src/tools/MLSToSlopes.hpp

```diff
@@ -0,0 +1,93 @@
+//
+// Copyright (c) 2015-2017, Deutsches Forschungszentrum für Künstliche Intelligenz GmbH.
+// Copyright (c) 2015-2017, University of Bremen
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// * Redistributions of source code must retain the above copyright notice, this
+//   list of conditions and the following disclaimer.
+//
+// * Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+#ifndef __MAPS_MLS_TO_SLOPES_HPP_
+#define __MAPS_MLS_TO_SLOPES_HPP_
+
+#include "../grid/GridMap.hpp"
+#include "../grid/MLSMap.hpp"
+
+namespace maps { namespace tools 
+{
+    /** This operator computes local slopes on a MLS map
+     *
+     * It acts on an MLSMapKalman and updates a GridMap<float> with the maximum local
+     * slope angles in radians.
+     * It also returns the absolute value of the maximal step between each cell and its direct
+     * neighbours.
+     *
+     * The default operation will compute the maximum slope between the topmost
+     * surfaces of a MLS. In practice, it means that it works only on MLS that
+     * have one patch per cell.
+     *
+     * It can be customized by subclassing and overloading the computeGradient
+     * operator
```

Comment:
If you want support this the computeGradient method must be virtual.
It seems this method is not implemented or used?

Reply:
Yes. That was also the case in the original envire version.

Reply:
Lets remove this method and the comment then. It was probably replaced by the PlaneFitting